FrameworkSystem/private/standardLogging/Logging.py

```diff
@@ -124,21 +124,22 @@ def _setOption(self, optionName, value, directCall=True):
     :params directCall: boolean indicating if it is a call by the user or not
     """
     # lock to prevent that two threads change the options at the same time
-    with self._lockOptions:
-      if self._optionsModified[optionName] and not directCall:
-        return
+    self._lockOptions.acquire()
```

Comment:
and I think you want to wrap all what comes ofter the acquire in a try block, and put the release in a ```finally```